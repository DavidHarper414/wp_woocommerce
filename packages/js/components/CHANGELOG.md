# Changelog 

This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [10.2.0](https://www.npmjs.com/package/@woocommerce/packages/js/components/v/10.2.0) - 2022-06-28 

-   Minor - Add step name to tour kit step type and export CloseHandler type to be reused elsewhere
-   Minor - Updated @automattic/tour-kit to 1.1.1 which has live resize functionality
-   Patch - Tweak tour kit gap between content and controls

## [10.1.0](https://www.npmjs.com/package/@woocommerce/packages/js/components/v/10.1.0) - 2022-06-09 

-   Minor - Add tour kit component
-   Minor - Update dependency `memoize-one` to ^6.0.0. #32936
-   Minor - Update dependency `react-dates` to ^21.8.0. #32883
-   Minor - Add Jetpack Changelogger
-   Minor - Update dependency `@wordpress/hooks` to ^3.5.0
-   Minor - Update dependency `@wordpress/icons` to ^8.1.0
-   Minor - Add `className` prop for Pill component. #32605
-   Patch - Removed unused react-router-dom dependency #33156
-   Patch - Standardize lint scripts: add lint:fix
-   Patch - Fix documentation for `TableCard` component
-   Patch - Update `StepperProps` prop types. #32712

---

[See legacy changelogs for previous versions](https://github.com/woocommerce/woocommerce-admin/blob/main/packages/components/CHANGELOG.md).
