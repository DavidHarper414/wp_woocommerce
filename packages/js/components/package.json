{
	"name": "@woocommerce/components",
	"version": "12.2.0",
	"description": "UI components for WooCommerce.",
	"author": "Automattic",
	"license": "GPL-3.0-or-later",
	"engines": {
		"node": "^16.14.1",
		"pnpm": "^8.6.7"
	},
	"keywords": [
		"wordpress",
		"woocommerce",
		"components"
	],
	"homepage": "https://github.com/woocommerce/woocommerce/tree/trunk/packages/js/components/README.md",
	"repository": {
		"type": "git",
		"url": "https://github.com/woocommerce/woocommerce.git"
	},
	"bugs": {
		"url": "https://github.com/woocommerce/woocommerce/issues"
	},
	"main": "build/index.js",
	"module": "build-module/index.js",
	"react-native": "src/index",
	"sideEffects": [
		"build-style/**",
		"src/**/*.scss"
	],
	"types": "build-types",
	"dependencies": {
		"@automattic/calypso-color-schemes": "^2.1.1",
		"@automattic/interpolate-components": "^1.2.1",
		"@automattic/tour-kit": "^1.1.1",
		"@types/wordpress__block-editor": "7.0.0",
		"@types/wordpress__block-library": "2.6.1",
		"@types/wordpress__blocks": "11.0.7",
		"@types/wordpress__components": "^19.10.5",
		"@types/wordpress__rich-text": "3.4.6",
		"@woocommerce/csv-export": "workspace:*",
		"@woocommerce/currency": "workspace:*",
		"@woocommerce/data": "workspace:*",
		"@woocommerce/date": "workspace:*",
		"@woocommerce/navigation": "workspace:*",
		"@wordpress/a11y": "wp-6.0",
		"@wordpress/api-fetch": "wp-6.0",
		"@wordpress/base-styles": "wp-6.0",
		"@wordpress/block-editor": "^9.8.0",
		"@wordpress/block-library": "^7.19.0",
		"@wordpress/blocks": "^11.21.0",
		"@wordpress/components": "wp-6.0",
		"@wordpress/compose": "wp-6.0",
		"@wordpress/core-data": "wp-6.0",
		"@wordpress/date": "wp-6.0",
		"@wordpress/deprecated": "wp-6.0",
		"@wordpress/dom": "wp-6.0",
		"@wordpress/element": "wp-6.0",
		"@wordpress/hooks": "wp-6.0",
		"@wordpress/html-entities": "wp-6.0",
		"@wordpress/i18n": "wp-6.0",
		"@wordpress/icons": "wp-6.0",
		"@wordpress/keyboard-shortcuts": "wp-6.0",
		"@wordpress/keycodes": "wp-6.0",
		"@wordpress/media-utils": "wp-6.0",
		"@wordpress/rich-text": "wp-6.0",
		"@wordpress/url": "wp-6.0",
		"@wordpress/viewport": "^4.20.0",
		"classnames": "^2.3.2",
		"core-js": "^3.34.0",
		"d3-axis": "^1.0.12",
		"d3-format": "^1.4.5",
		"d3-scale": "^2.2.2",
		"d3-scale-chromatic": "^1.5.0",
		"d3-selection": "^1.4.2",
		"d3-shape": "^1.3.7",
		"d3-time-format": "^2.3.0",
		"dompurify": "^2.4.7",
		"downshift": "^6.1.12",
		"emoji-flags": "^1.3.0",
		"gridicons": "^3.4.2",
		"memoize-one": "^6.0.0",
		"moment": "^2.29.4",
		"moment-timezone": "^0.5.43",
		"prop-types": "^15.8.1",
		"react-dates": "^21.8.0",
		"react-transition-group": "^4.4.5"
	},
	"peerDependencies": {
		"@types/react": "^17.0.71",
		"@types/react-dom": "^17.0.25",
		"@wordpress/data": "wp-6.0",
		"lodash": "^4.17.0",
		"react": "^17.0.2",
		"react-dom": "^17.0.2"
	},
	"publishConfig": {
		"access": "public"
	},
	"devDependencies": {
		"@babel/core": "^7.23.5",
		"@babel/runtime": "^7.23.5",
		"@storybook/addon-actions": "6.5.17-alpha.0",
		"@storybook/addon-console": "^1.2.3",
		"@storybook/addon-controls": "6.5.17-alpha.0",
		"@storybook/addon-docs": "6.5.17-alpha.0",
		"@storybook/addon-knobs": "^6.4.0",
		"@storybook/addon-links": "6.5.17-alpha.0",
		"@storybook/addons": "6.5.17-alpha.0",
		"@storybook/api": "6.5.17-alpha.0",
		"@storybook/components": "6.5.17-alpha.0",
		"@storybook/core-events": "6.5.17-alpha.0",
		"@storybook/react": "6.5.17-alpha.0",
		"@storybook/theming": "6.5.17-alpha.0",
		"@testing-library/dom": "8.11.3",
		"@testing-library/jest-dom": "5.16.2",
		"@testing-library/react": "12.1.3",
		"@testing-library/react-hooks": "8.0.1",
		"@testing-library/user-event": "13.5.0",
		"@types/jest": "^27.5.2",
		"@types/lodash": "^4.14.202",
		"@types/prop-types": "^15.7.11",
		"@types/testing-library__jest-dom": "^5.14.9",
		"@types/uuid": "^8.3.4",
		"@types/wordpress__components": "^19.10.5",
		"@types/wordpress__data": "6.0.0",
		"@types/wordpress__media-utils": "3.0.0",
		"@types/wordpress__viewport": "2.5.4",
		"@woocommerce/eslint-plugin": "workspace:*",
		"@woocommerce/internal-js-tests": "workspace:*",
		"@woocommerce/internal-style-build": "workspace:*",
		"@wordpress/babel-preset-default": "^6.17.0",
		"@wordpress/browserslist-config": "wp-6.0",
		"concurrently": "^7.6.0",
		"css-loader": "^3.6.0",
		"eslint": "^8.55.0",
		"jest": "~27.5.1",
		"jest-cli": "~27.5.1",
		"postcss": "^8.4.32",
		"postcss-loader": "^4.3.0",
		"react": "^17.0.2",
		"rimraf": "^3.0.2",
		"sass-loader": "^10.5.0",
		"ts-jest": "~29.1.1",
		"typescript": "^5.3.3",
		"uuid": "^8.3.2",
		"webpack": "^5.89.0",
		"webpack-cli": "^3.3.12"
	},
	"scripts": {
		"turbo:build": "pnpm run build:js && pnpm run build:css",
		"turbo:test": "jest --config ./jest.config.json",
		"prepare": "composer install",
		"changelog": "composer exec -- changelogger",
		"build": "pnpm -w exec turbo run turbo:build --filter=$npm_package_name",
		"test": "pnpm test:js",
		"test:js": "pnpm -w exec turbo run turbo:test --filter=$npm_package_name",
		"lint": "eslint --ext=js,ts,tsx src",
		"build:js": "tsc --project tsconfig.json && tsc --project tsconfig-cjs.json",
		"build:css": "webpack",
		"clean": "pnpm exec rimraf tsconfig.tsbuildinfo build build-*",
		"lint:fix": "eslint --ext=js,ts,tsx src --fix",
		"prepack": "pnpm run clean && pnpm run build",
		"start": "concurrently \"tsc --project tsconfig.json --watch\" \"tsc --project tsconfig-cjs.json --watch\" \"webpack --watch\"",
		"test:update-snapshots": "pnpm run test -- --updateSnapshot",
		"test-staged": "jest --bail --config ./jest.config.json --findRelatedTests"
	},
	"lint-staged": {
		"*.(t|j)s?(x)": [
			"pnpm lint:fix",
			"pnpm test-staged"
		]
	}
}
