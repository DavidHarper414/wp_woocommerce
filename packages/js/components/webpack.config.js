/**
 * Internal dependencies
 */
const { webpackConfig } = require( '@woocommerce/internal-style-build' );

/**
 * External dependencies
 */
const path = require( 'path' );

module.exports = {
	mode: process.env.NODE_ENV || 'development',
	cache: ( process.env.CI && { type: 'memory' } ) || {
		type: 'filesystem',
		cacheDirectory: path.resolve(
			__dirname,
			'../../../node_modules/.cache/webpack-components'
		),
	},
	entry: {
		'build-style': __dirname + '/src/style.scss',
	},
	output: {
		path: __dirname,
	},
	module: {
		parser: webpackConfig.parser,
		rules: webpackConfig.rules,
	},
	plugins: webpackConfig.plugins,
};
