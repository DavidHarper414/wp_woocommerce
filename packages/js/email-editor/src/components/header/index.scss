// Document actions - Component in header for displaying email/campaign title edit popup
.mailpoet-email-editor-campaign-name {
	align-items: center;
	display: flex;
	flex-direction: row;
	justify-content: center;
	min-width: 0;

	.components-dropdown {
		display: inline-flex;
	}

	.components-button {
		min-width: 0;
		padding: 0;
	}

	.mailpoet-email-campaign-name-link {
		display: inline-flex;
		height: fit-content;
		margin-right: 10px;
		margin-top: 10px;
	}
}

// Document actions - Popup for editing email/campaign title
.mailpoet-email-editor-campaign-name-dropdown {

	/* stylelint-disable-next-line selector-class-pattern */
	.components-popover__content {
		min-width: 280px;
		padding: 0;
	}

	.mailpoet-email-editor-email-title-edit {
		padding: 16px;
	}
}

.mailpoet-email-editor-save-button-dropdown {

	/* stylelint-disable-next-line selector-class-pattern */
	.components-popover__content {
		min-width: 280px;
	}
}

// We currently don't support command palette so we hide the shortcut
/* stylelint-disable-next-line selector-class-pattern */
.editor-document-bar__shortcut {
	display: none !important;
}

// Temporarily hide text about amount of changes to save. The wording is not great for email context.
/* stylelint-disable-next-line selector-class-pattern */
.entities-saved-states__text-prompt {

	p {
		display: none;
	}
}

// Ensure checkbox controls in save panel dropdown to span the full width
.mailpoet-email-editor-save-button-dropdown
/* stylelint-disable-next-line selector-class-pattern */
.components-panel__row
/* stylelint-disable-next-line selector-class-pattern */
.components-checkbox-control {
	width: 100%;
}
