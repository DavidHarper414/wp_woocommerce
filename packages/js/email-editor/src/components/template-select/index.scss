/* Move the less specific selector before the more specific one */

/* stylelint-disable selector-class-pattern */
.block-editor-block-patterns-list__item .block-editor-block-preview__container {
	align-items: flex-start !important;
}

.block-editor-block-patterns-explorer
.block-editor-block-patterns-list
.block-editor-block-preview__container {
	max-height: 300px !important;
}
/* stylelint-enable */

/* stylelint-disable-next-line selector-class-pattern */
.block-editor-block-patterns-list__list-item {
	overflow: hidden;
}

.email-editor-recent-templates-info {
	background-color: #f0f6fc;
	margin-bottom: 1.5rem;
	padding: 0.5rem;

	path {
		fill: #007cba;
	}
}

/* stylelint-disable-next-line selector-class-pattern */
.email-editor-pattern__list-item {
	border: 1px solid #ddd;
	border-radius: 4px;

	/* stylelint-disable-next-line selector-class-pattern */
	.block-editor-block-preview__container::after {
		outline: unset !important;
		outline-offset: unset !important;
	}

	/* stylelint-disable-next-line selector-class-pattern */
	.block-editor-block-patterns-list__item-title {
		font-size: unset !important;
		padding-left: 1rem;
	}
}

.email-editor-modal-footer {
	background-color: #fff;
	bottom: 0;
	padding: 1rem 2rem;
	position: absolute;
	right: 0;
}
