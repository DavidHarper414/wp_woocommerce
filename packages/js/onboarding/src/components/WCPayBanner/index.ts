export * from './Icons';
export * from './WCPayBannerImage';
export * from './PaymentMethodsIcons';
export * from './WCPayBanner';
