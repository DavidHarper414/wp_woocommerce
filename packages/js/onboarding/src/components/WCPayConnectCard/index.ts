export * from './WCPayConnectCard';
export * from './Icons';
export * from './PaymentMethodsIcons';
