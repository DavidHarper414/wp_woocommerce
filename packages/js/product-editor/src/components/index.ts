export {
	ProductSectionLayout as __experimentalProductSectionLayout,
	ProductFieldSection as __experimentalProductFieldSection,
} from './product-section-layout';
export { WooProductFieldItem as __experimentalWooProductFieldItem } from './woo-product-field-item';
export { WooProductSectionItem as __experimentalWooProductSectionItem } from './woo-product-section-item';
export { WooProductTabItem as __experimentalWooProductTabItem } from './woo-product-tab-item';
export { DetailsNameField as __experimentalDetailsNameField } from './field-details-name';
export { DetailsFeatureField as __experimentalDetailsFeatureField } from './field-details-feature';
export {
	DetailsCategoriesField as __experimentalDetailsCategoriesField,
	CategoryField as __experimental__experimentalDetailsCategoryField,
} from './field-details-categories';
