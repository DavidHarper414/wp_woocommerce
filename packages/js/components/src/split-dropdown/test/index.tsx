/**
 * External dependencies
 */
import { render } from '@testing-library/react';
import { createElement } from '@wordpress/element';
import { Button } from '@wordpress/components';

/**
 * Internal dependencies
 */
import { SplitDropdown } from '..';

describe( 'SplitDropdown', () => {
	test( 'it renders correctly collapsed', () => {
		const { container, getByText, queryByText } = render(
			<SplitDropdown>
				<Button>Default Action</Button>
				<Button className="test-collapsed-button">
					Secondary Action
				</Button>
				<Button className="test-collapsed-button">
					Tertiary Action
				</Button>
			</SplitDropdown>
		);
		expect( container ).toMatchSnapshot();

		// should have correct content
		expect( getByText( 'Default Action' ) ).toBeInTheDocument();
		expect( queryByText( 'Secondary Action' ) ).not.toBeInTheDocument();

		// should have correct classes
		const presentClassNames = [
			'woocommerce-split-dropdown__container',
			'woocommerce-split-dropdown__main-button',
			'woocommerce-split-dropdown__toggle',
		];
		presentClassNames.map( ( className ) =>
			expect(
				container.getElementsByClassName( className )
			).toHaveLength( 1 )
		);
	} );
} );
