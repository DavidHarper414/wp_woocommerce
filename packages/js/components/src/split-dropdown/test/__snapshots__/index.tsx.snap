// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SplitDropdown it renders correctly collapsed 1`] = `
<div>
  <div
    class="components-button-group woocommerce-split-dropdown woocommerce-split-dropdown__container"
    role="group"
  >
    <button
      class="components-button woocommerce-split-dropdown__main-button is-primary"
      type="button"
    >
      Default Action
    </button>
    <div
      class="components-dropdown"
      tabindex="-1"
    >
      <button
        class="components-button woocommerce-split-dropdown__toggle is-primary"
        type="button"
      >
        <svg
          aria-hidden="true"
          focusable="false"
          height="24"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M17.5 11.6L12 16l-5.5-4.4.9-1.2L12 14l4.5-3.6 1 1.2z"
          />
        </svg>
      </button>
    </div>
  </div>
</div>
`;
